functional-samples/cookbook.geolocation-offscreen/service_worker.js

```diff
@@ -0,0 +1,80 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+const OFFSCREEN_DOCUMENT_PATH = '/offscreen.html';
+
+chrome.runtime.onMessage.addListener(handleMessages);
+
+let creating; // A global promise to avoid concurrency issues
+
+async function hasDocument() {
+  // Check all windows controlled by the service worker to see if one
+  // of them is the offscreen document with the given path
+  const offscreenUrl = chrome.runtime.getURL(OFFSCREEN_DOCUMENT_PATH);
```

Comment:
In service workers, you can access extension resources directly. the `runtime.getURL()` was created to access extension assets in content scripts.

Reply:
removed!